content/shell/renderer/shell_content_renderer_client.cc

```diff
@@ -270,6 +271,55 @@ ShellContentRendererClient::CreateURLLoaderThrottleProvider(
   return std::make_unique<ShellContentRendererUrlLoaderThrottleProvider>();
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+media::SupportedCodecs GetStarboardEmeSupportedCodecs() {
+  media::SupportedCodecs codecs = ::media::EME_CODEC_AAC | ::media::EME_CODEC_AVC1 |
+                           ::media::EME_CODEC_VP9_PROFILE0 |
+                           ::media::EME_CODEC_VP9_PROFILE2 |
+                           ::media::EME_CODEC_VP8 |
+                           ::media::EME_CODEC_OPUS |
+                           ::media::EME_CODEC_AV1 |
+                           ::media::EME_CODEC_AC3 |
+                           ::media::EME_CODEC_EAC3;
+  return codecs;
+}
+
+void AddStarboardCmaKeySystems(::media::KeySystemInfos* key_system_infos) {
+  media::SupportedCodecs codecs = GetStarboardEmeSupportedCodecs();
+
+  using Robustness = cdm::WidevineKeySystemInfo::Robustness;
+
+  const base::flat_set<media::EncryptionScheme> kEncryptionSchemes = {
+    media::EncryptionScheme::kCenc, media::EncryptionScheme::kCbcs};
+
+  const base::flat_set<media::CdmSessionType> kSessionTypes = {
+    media::CdmSessionType::kTemporary, media::CdmSessionType::kPersistentLicense};
+
+  key_system_infos->emplace_back(new cdm::WidevineKeySystemInfo(
+      codecs,                             // Regular codecs.
+      kEncryptionSchemes,                 // Encryption schemes.
+      kSessionTypes,                      // Session types.
+      codecs,                             // Hardware secure codecs.
+      kEncryptionSchemes,                 // Hardware secure encryption schemes.
+      kSessionTypes,                      // Hardware secure session types.
+      Robustness::HW_SECURE_CRYPTO,       // Max audio robustness.
+      Robustness::HW_SECURE_ALL,          // Max video robustness.
+      media::EmeFeatureSupport::ALWAYS_ENABLED,  // Persistent state.
+      media::EmeFeatureSupport::ALWAYS_ENABLED));  // Distinctive identifier.
+}
+
+void ShellContentRendererClient::GetSupportedKeySystems(
+    media::GetSupportedKeySystemsCB cb) {
+  media::KeySystemInfos key_systems;
+  if (base::FeatureList::IsEnabled(media::kExternalClearKeyForTesting))
+    key_systems.push_back(
+        std::make_unique<cdm::ExternalClearKeyKeySystemInfo>());
+  AddStarboardCmaKeySystems(&key_systems);
+  std::move(cb).Run(std::move(key_systems));
+}
+
+#else // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 #if BUILDFLAG(ENABLE_MOJO_CDM)
```

Comment:
Is it possible for ENABLE_MOJO_CDM to ever be turned on for our builds?

Reply:
Unlikely, I haven't heard of any talk of running our DRM system in a separate process.

Reply:
If that's the case, then I don't see the reason to have this be an #else block instead of just putting an #endif above it?

Reply:
Did some experiments, and it is actually enabled by default. 
I disabled it in media/mojo/BUILD.gn but it then requires some other media changes. Rather than bring those changes into this PR, i've opened a big to track doing that separately - b/378740140

Reply:
Sounds good